Add WAFS wind direction verification to EVSv2.0

YaliMao-NOAA opened the pull request:
<b>Note to developers: You must use this PR template!</b>

## Description of Changes

> Please include a summary of the changes and the related GitHub issue(s). Please also include relevant motivation and context.

Add wind direction verification to WAFS STATs files.
1. Use VCNT line type for wind direction per discussion under [the MET issue #2395 ](https://github.com/dtcenter/MET/issues/2395)opened by Yali Mao in Jan 2023, and [another related MET issue #2590](https://github.com/dtcenter/METplus/discussions/2590) 
2. Job card needs to add one more CPU to process wind direction verification through MPMD, no significant impact on running time. Please refer to the update of dev/drivers/scripts/stats/wafs/jevs_wafs_atmos_stats.sh
3. Parm files under parm/metplus_config/stats/wafs/atmos are renamed for adding wind direction
4. The WAFS file name under script/ scripts/stats is renamed according to EE2 standards, now it's wafs/exevs_wafs_stats.sh.

To verify the STAT result, the WAFS plotting scripts are updated too though it's not mandatory in EVS operations.
1. Job card uses 40 CPUs after the code re-constructed.
2. Add one contingency table for each icing severity ROC plotting

## Developer Questions and Checklist
* Is this a high priority PR? If so, why and is there a date it needs to be merged by?
* Do you have any planned upcoming annual leave/PTO? Yes, the week of July 4, 2025.
* Are there any changes needed in the times when the jobs are supposed to run/kick-off?
  
- [x ] The code changes follow [NCO's EE2 Standards](https://www.nco.ncep.noaa.gov/idsb/implementation_standards/ImplementationStandards.v11.0.0.pdf).
- [ ] Developer's name is removed throughout the code and have used `${USER}` where necessary throughout the code.
- [ ] References the feature branch for `HOMEevs` are removed from the code.
- [ ] J-Job environment variables, COMIN and COMOUT directories, and output follow what has been [defined](https://docs.google.com/document/d/1JWg_4q80aYmmAoD21GFjp9R9y5-3w7WGM3-0HJk0Pjs/edit#heading=h.7ysbr191vzu4) for EVS.
- [ ] Jobs over 15 minutes in runtime have restart capability.
- [ ] If applicable, changes in the `dev/drivers/scripts` or `dev/modulefiles` have been made in the corresponding `ecf/scripts` and `ecf/defs/evs-nco.def`? 
- [ ] Jobs contain the appropriate file checking and don't run METplus for any missing data.
- [ ] Code is using METplus wrappers structure and not calling MET executables directly.
- [ ] Log is free of any ERRORs or WARNINGs.

## Testing Instructions

> Please include testing instructions for the PR assignee. Include all relevant input datasets needed to run the tests.
1. cd $HOMEevs
2. qsub dev/drivers/scripts/stats/wafs/jevs_wafs_atmos_stats.sh
3. The test outputs are saved under /lfs/h2/emc/vpppg/noscrub/yali.mao/evs/v2.0/stats/wafs on Dogwood
4. To verify no impact on the existing icing and uvt verification, I changed versions/run.ver to evs_ver=v1.0.0 and met_ver=11.0.2, the outputs are saved under /lfs/h2/emc/vpppg/noscrub/yali.mao/evs/v1.0/stats/wafs on Dogwood, there are no difference between the test results and the operational ones.

Commit:
- Rename EVS WAFS script names according to the job names
- Correct some comments

Commit:
Add wind direction verification to WAFS EVS stat

Commit:
Modify parm files for better code organizing for wind direction verification.

Commit:
Change EVS version back to v2

Commit:
Add contigency table for icing severity ROC verification

Commit:
Add wind direction verification

Commit:
Merge remote-tracking branch 'upstream/develop' into wdir

Commit:
Create a tarball for contingency table in html for WAFS icing severity

AliciaBentley-NOAA commented on dev/drivers/scripts/plots/wafs/jevs_wafs_atmos_plots.sh:
```diff
@@ -1,10 +1,10 @@
-#PBS -N jevs_wafs_atmos_plots
+#PBS -N jevs_wafs_plots
```
@YaliMao-NOAA The PBS job name needs to match the name of the script (e.g., jevs_wafs_atmos_plots). Can you please add "atmos" back to the PBS job names? This will match the operational wafs stats job, which we don't want to change (see this [ops .ecf script](https://github.com/NOAA-EMC/EVS/blob/develop/ecf/scripts/stats/wafs/jevs_wafs_atmos_stats.ecf#L1C9-L1C29)).

YaliMao-NOAA replied:
@AliciaBentley-NOAA Back in Oct 2024, Mallory changed the WAFS job file names. https://github.com/NOAA-EMC/EVS/commit/8d2db699867cc10a76d8d998011d74691b6b3784
May I double confirm that I change the file names back?

AliciaBentley-NOAA replied:
@YaliMao-NOAA For more complicated components of EVS like `global_det`, $RUN can be several different things (e.g., atmos, wave). Here are two examples where $RUN is different in the dev/driver script name. 

jevs_global_det_gfs_atmos_grid2obs_stats.sh
jevs_global_det_gfs_wave_grid2obs_stats.sh

These stats jobs both use the same J-job, which does **_not_** have $RUN in the name: **JEVS_GLOBAL_DET_STATS**

The edit that Mallory made removed $RUN from the J-job name, but left it in the dev/driver names, etc. Please add "atmos" back into the job names so that the job name matches the name of the dev/drivers and .ecf scripts. 

#PBS -N jevs_wafs_**atmos**_plots

YaliMao-NOAA replied:
@AliciaBentley-NOAA Sorry I am on vacation and can’t make changes now.  Do you mean the changes to be done is the job cards under dev/driver? The file names under jobs/ and scripts m/ are fine? From what I learned from EIB, developers have much freedom to do under dev/ because all scripts under dev/ will not go operational, they are just references to the code manager(s) for their own version of job cards for code delivery. I hope it doesn’t affect further testing. I will make the changes after I get back to work next week.

AliciaBentley-NOAA commented:
@YaliMao-NOAA Enjoy your vacation! We can change this PR back to match the other EVS components and merged into develop when you get back! Thanks!

Commit:
Per Alicia's suggestion, change the job card names under dev/drivers from 8d2db69
back to be consistent with the operational.

Commit:
Merge remote-tracking branch 'upstream/develop' into wdir

YaliMao-NOAA commented:
@AliciaBentley-NOAA I changed the job card names to jevs_wafs_atmos_stats and jevs_wafs_atmos_plots respectively, but leaving the job names without ATMOS: JEVS_WAFS_STATS and JEVS_WAFS_PLOTS, leaving the script names without atmos either. Please let me know whether they are correct.

AliciaBentley-NOAA commented on dev/drivers/scripts/plots/wafs/jevs_wafs_atmos_plots.sh:
```diff
@@ -45,12 +45,12 @@ export COMPONENT=wafs
-#export COMIN=${COMIN:-/lfs/h2/emc/vpppg/noscrub/${USER}/${NET}/$evs_ver_2d}
-export COMIN=/lfs/h1/ops/prod/com/${NET}/${evs_ver_2d}
+export COMIN=${COMIN:-/lfs/h2/emc/vpppg/noscrub/${USER}/${NET}/$evs_ver_2d}
+#export COMIN=/lfs/h1/ops/prod/com/${NET}/${evs_ver_2d}
```
@YaliMao-NOAA Please remove this commented out line in your dev driver. Thanks!

AliciaBentley-NOAA commented on dev/drivers/scripts/plots/wafs/jevs_wafs_atmos_plots.sh:
```diff
@@ -45,12 +45,12 @@ export COMPONENT=wafs
 export COMROOT=/lfs/h2/emc/ptmp/$USER
 
-export DAYS_LIST=${DAYS_LIST:-"90 31"}
+export DAYS_LIST=${DAYS_LIST:-"90"}
```
@YaliMao-NOAA Please update the name of this script from `jevs_wafs_atmos_plots.sh` to `jevs_wafs_atmos_plots_last90days.sh` based on this change to just plot the last 90 days. Thanks!

YaliMao-NOAA replied:
The script is designed for plotting 90,31 and any days longer than 90 days.

AliciaBentley-NOAA replied:
Thanks! We don't need to change the name of this script if it does 90 and 5+ years (and this is in dev only).

AliciaBentley-NOAA commented on dev/drivers/scripts/stats/wafs/jevs_wafs_atmos_stats.sh:
```diff
@@ -1,10 +1,10 @@
 #PBS -A VERF-DEV
 #PBS -l walltime=00:25:00
-#PBS -l place=shared,select=1:ncpus=4:mem=10GB
+#PBS -l place=shared,select=1:ncpus=5:mem=10GB
```
@YaliMao-NOAA Please update the job settings in the WAFS stats .ecf script to match those in this dev driver. The job name and requested nodes may need to be updated.

YaliMao-NOAA replied:
The extra CPU is requested for wind direction.

AliciaBentley-NOAA replied:
@YaliMao-NOAA Yep, that's fine. Please make sure that the nodes in the .ecf that matches this dev driver are also updated. NCO runs the .ecf script, and does not touch the dev drivers. Thanks!

YaliMao-NOAA replied:
@AliciaBentley-NOAA This reminds me that I need to add ATMOS back to the job name in jevs_wafs_atmos_stats.ecf, literally to revert all the changes in https://github.com/NOAA-EMC/EVS/commit/8d2db699867cc10a76d8d998011d74691b6b3784 which were made by Mallory last year.

AliciaBentley-NOAA replied:
@YaliMao-NOAA Looking at the commit you just shared, it looks like the only thing that Mallory changed was the name of the J-Jobs by removing the atmos. I agree with this change and leaving the other scripts the same. This PR can just add wind direction without changing any other script names. We can discuss how to back out the extra changes that you made in this PR later today.

AliciaBentley-NOAA replied:
@YaliMao-NOAA I can go over the EVS scripts structure with you during today's meeting. There are reasons why the J-Jobs don't have atmos in them and the dev drivers, .ecf scripts, and scripts still do. This is consistent with other ops EVS components and will hopefully make more sense later today.

AliciaBentley-NOAA commented on jobs/JEVS_WAFS_PLOTS:
```diff
@@ -89,8 +89,8 @@ export USH_DIR=${USH_DIR:-$HOMEevs/ush/$COMPONENT}
-export DAYS_LIST=${DAYS_LIST:-"90 31"}
-$SCRIPTSevs/exevs_wafs_atmos_plots.sh
+export DAYS_LIST=${DAYS_LIST:-"90"}
+$SCRIPTSevs/exevs_wafs_plots.sh
```
@YaliMao-NOAA The name of this script needs to match the name of the .ecf script and dev driver. Please add the atmos back. Thanks!

YaliMao-NOAA replied:
I will do

AliciaBentley-NOAA commented on jobs/JEVS_WAFS_STATS:
```diff
@@ -93,7 +93,7 @@ export MASKS=${MASKS:-$FIXevs/masks}
 # Execute the script using MPMD
 ############################################
-$SCRIPTSevs/exevs_wafs_atmos_stats.sh
+$SCRIPTSevs/exevs_wafs_stats.sh
```
@YaliMao-NOAA The name of this script needs to match the name of the .ecf script and dev driver. Please add the atmos back. Thanks!

YaliMao-NOAA replied:
I will do

AliciaBentley-NOAA commented on parm/metplus_config/stats/wafs/atmos/GridStat_fcstWAFS_obsGCIPicesev.conf:
```diff
@@ -62,6 +62,9 @@ GRID_STAT_INTERP_TYPE_WIDTH = 1
 GRID_STAT_NEIGHBORHOOD_WIDTH = 1
 GRID_STAT_NEIGHBORHOOD_SHAPE = SQUARE
 GRID_STAT_NEIGHBORHOOD_COV_THRESH = >=0.5
+# In METplusv6/METv12, GRID_STAT_GRID_WEIGHT_FLAG = COS_LAT is default to CTC line type.
```
@YaliMao-NOAA MET/METplus version numbers can not be anywhere in EVS beside run.ver (not even in a comment). You can either remove this comment or remove the version numbers contained within the comment. Thanks!

YaliMao-NOAA replied:
@AliciaBentley-NOAA I left the comment because it is the new feature beginning METplusv6/METv12. It helps understanding the code change.

AliciaBentley-NOAA replied:
@YaliMao-NOAA Thanks. NCO operational standards require that version numbers outside run.ver be removed. Please remove these version numbers before this is flagged by EIB during their EE2 review or by NCO after code delivery.

CC @PerryShafran-NOAA @ShannonShields-NOAA

YaliMao-NOAA replied:
@AliciaBentley-NOAA Thank you for the meeting. I made the changes as advised, leaving j-job file names without 'ATMOS', while keeping 'atmos' in other scripts.

Commit:
Per Alicia's clarification, WAFS EVS leaves j-job file names without 'ATMOS',
while keeping 'atmos' in dev/driver, ecf/scripts, and scripts

AliciaBentley-NOAA commented on dev/drivers/scripts/plots/wafs/jevs_wafs_atmos_plots.sh:
```diff
@@ -4,7 +4,7 @@
 #PBS -A VERF-DEV
 #PBS -l walltime=01:00:00
-#PBS -l place=shared,select=1:ncpus=60:mem=200GB
+#PBS -l place=shared,select=1:ncpus=40:mem=200GB
```
@YaliMao-NOAA Is there a matching .ecf script for this plots dev driver? If so, can the ncpus be updated in that .ecf script? If not, that's OK for now. I know the plotting scripts won't be in prod for EVS v2.0.

YaliMao-NOAA replied:
No ecf script for plots. I think the reason is because WAFS EVS plotting is not in prod.

AliciaBentley-NOAA commented on ush/wafs/evs_wafs_atmos_plot_util.py:
```diff
@@ -1384,8 +1468,9 @@ def calculate_stat(logger, model_data, stat):
-   elif stat == 'pcor':
+   elif stat == 'pcor' or stat == 'corr':
       if line_type == 'SL1L2':
+         #var_f = ffbar - fbar*obar
```
@YaliMao-NOAA This commented out line looks a lot like the line below it. Can this commented out line be removed?

YaliMao-NOAA replied:
I include this in the next commit.

AliciaBentley-NOAA commented on ush/wafs/evs_wafs_atmos_plot_util.py:
```diff
@@ -1509,6 +1606,21 @@ def calculate_stat(logger, model_data, stat):
+       stat_values = dir_rmse
    else:
       logger.error(stat+" is not a valid option")
       exit(1)
```
@YaliMao-NOAA Adding a note to myself to remember to remove the wafs plot ush/ scripts (as well as the dev driver, etc.) from the EVS v2.0 code delivery.

AliciaBentley-NOAA commented on ush/wafs/evs_wafs_atmos_stats_preparedata.sh:
```diff
@@ -203,6 +203,10 @@ fi
-    ${METPLUS_PATH}/ush/run_metplus.py -c $MACHINE_CONF -c $PARMevs/StatAnalysis_fcstWAFS_obs${OBSERVATION}_GatherbyDay.conf
+    #INPUT_BASE = {ENV[STATSOUTsmall]}=$STATSDIR/$RUN.$VDATE
+    #STAT_ANALYSIS_OUTPUT_DIR is defined in config.evs.wafs.standalone, and created & used by StatAnalysis_fcstWAFS*
+    ${METPLUS_PATH}/ush/run_metplus.py -c $MACHINE_CONF -c $PARMevs/StatAnalysis_fcstWAFS_obs${OBSERVATION}${VAR}_GatherbyDay.conf
```
@YaliMao-NOAA Please clean up the commented out lines in this section. Thanks!

YaliMao-NOAA replied:
@AliciaBentley-NOAA I am removing the first line of comment. Can I leave the second line of comment because it helps me understanding, developing and debugging the code?

AliciaBentley-NOAA replied:
@YaliMao-NOAA Yes, comments that help developers and NCO understand what the code does are totally fine to keep! Only commented-out lines of code that are no longer used need to be removed. Thank you for asking!

AliciaBentley-NOAA commented:
@YaliMao-NOAA I have started testing your WAFS stats job! You can find the information about my test below. The stats job is still running, so please check back again when the job completes to assess the log file and the stats output. Thanks!

### jevs_wafs_atmos_stats.sh
**Log file:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/pr774test/EVS/dev/drivers/scripts/stats/wafs/jevs_wafs_atmos_stats.o209928125
**DATA:** /lfs/h2/emc/stmp/alicia.bentley/evs_test/prod/tmp/jevs_wafs_atmos_stats.209928125.cbqs01/
**Stats:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/evs/v2.0/stats/wafs/

CC @ShannonShields-NOAA @SamiraArdani-NOAA

YaliMao-NOAA commented:
> @YaliMao-NOAA I have started testing your WAFS stats job! You can find the information about my test below. The stats job is still running, so please check back again when the job completes to assess the log file and the stats output. Thanks!
> 
> ### jevs_wafs_atmos_stats.sh
> **Log file:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/pr774test/EVS/dev/drivers/scripts/stats/wafs/jevs_wafs_atmos_stats.o209928125 **DATA:** /lfs/h2/emc/stmp/alicia.bentley/evs_test/prod/tmp/jevs_wafs_atmos_stats.209928125.cbqs01/ **Stats:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/evs/v2.0/stats/wafs/
> 
> CC @ShannonShields-NOAA @SamiraArdani-NOAA

@AliciaBentley-NOAA Thank you for the fast testing. The log file looks good. The stat outputs (icesev and uvt) are bitwise identical to the operational other than the METplus version numbers. The wdir stat output looks good.

AliciaBentley-NOAA commented:
@YaliMao-NOAA  Thanks! I agree that the log file looks good. There are no errors or warnings. The requested walltime meets NCO standards (see below). I will move onto plots now!

```
    resources_used.walltime = 00:11:05
    ctime = Tue Jul  8 14:48:18 2025
    mtime = Tue Jul  8 15:00:12 2025
    Resource_List.walltime = 00:25:00
```

AliciaBentley-NOAA commented:
@YaliMao-NOAA Should I point COMIN to `/lfs/h2/emc/vpppg/noscrub/yali.mao/evs/v2.0/stats/wafs` to test the plots job? Otherwise, I don't think we'll have any wind direction stats to plot! Please let me know. Thanks!

Commit:
Remove commented-out lines that are not used anymore.

YaliMao-NOAA commented:
@AliciaBentley-NOAA Yes, you can. I just transferred 3 days of data from Dogwood to Cactus /lfs/h2/emc/vpppg/noscrub/yali.mao/evs/v2.0/stats/wafs. I am going to copy your output to my folder too

AliciaBentley-NOAA commented:
@YaliMao-NOAA I have kicked off the WAFS plots job and pointed COMIN to your stat files. You can find the information about the job at the paths below. Please check back when the job completes to review the log file and output. Please also share a link to the plots that are produced so that we can view them online. Thanks!

### jevs_wafs_atmos_plots.sh
**Log file:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/pr774test/EVS/dev/drivers/scripts/plots/wafs/jevs_wafs_atmos_plots.o209931534
**DATA:** /lfs/h2/emc/stmp/alicia.bentley/evs_test/prod/tmp/jevs_wafs_atmos_plots.209931534.cbqs01/
**Plots:** /lfs/h2/emc/ptmp/alicia.bentley/evs/v2.0/plots/wafs/

CC @ShannonShields-NOAA @SamiraArdani-NOAA

YaliMao-NOAA commented:
@AliciaBentley-NOAA Here is the plotting link: https://www.emc.ncep.noaa.gov/users/verification/aviation/wafs/para/. Only 90 days is available.

AliciaBentley-NOAA commented:
@YaliMao-NOAA Thanks! I'll finish reviewing your PR today. Please sync your fork (five PRs have been merged in the last few days).

CC @ShannonShields-NOAA @SamiraArdani-NOAA

AliciaBentley-NOAA commented:
@YaliMao-NOAA I am seeing a lot of lines with "failed" and "WARNING" in the plots log files. Can you please investigate these error messages? Note that symbolic links are not allowed in ops code, but we are currently keeping WAFS plots in dev. 

```
ln: failed to create symbolic link '/lfs/h2/emc/stmp/alicia.bentley/evs_test/prod/tmp/jevs_wafs_atmos_plots.209931534.cbqs01/datainput/GFS_1P25_uvt/./evs.stats.wafs.atmos.grid2grid_uvt1p25.v20250623.stat'
ln: failed to create symbolic link '/lfs/h2/emc/stmp/alicia.bentley/evs_test/prod/tmp/jevs_wafs_atmos_plots.209931534.cbqs01/datainput/GFS_1P25_uvt/./evs.stats.wafs.atmos.grid2grid_uvt1p25.v20250623.stat': File exists

07/08 15:41:47.833 (evs_wafs_atmos_plot_roc_curve.py:501) WARNING: blend data were all NaNs and will not be plotted.
07/08 15:41:47.833 (evs_wafs_atmos_plot_roc_curve.py:507) WARNING: Could not find (and cannot plot) farate and/or pod stats for ICESEV at any threshold. 
07/08 15:41:47.833 (evs_wafs_atmos_plot_roc_curve.py:511) WARNING: This may occur if no forecast or observed events were counted at any threshold for any model, so that all performance statistics are undefined. Continuing ...
```

CC @ShannonShields-NOAA @SamiraArdani-NOAA

YaliMao-NOAA commented:
@AliciaBentley-NOAA I will take a look.

AliciaBentley-NOAA commented:
@YaliMao-NOAA Any update on the failures and WARNING messages that we saw in your log files?

CC @ShannonShields-NOAA @SamiraArdani-NOAA

YaliMao-NOAA commented:
@AliciaBentley-NOAA No update yet. I will take a look from this Thursday.

YaliMao-NOAA commented:
@AliciaBentley-NOAA Those warnings come from no icing over some subregions (there are 10 subregions in WAFS EVS) on some vertical levels, considering it's summer time and the plotting is over a few days.
Do you want to get rid of the warnings completely or just to check the reasons of the warnings?

AliciaBentley-NOAA commented:
@YaliMao-NOAA Thanks! It's OK to have WARNING messages when there are not stats, as long as the WARNINGs are clear. Since this is currently not code planned for ops, I think that these WARNINGs are OK for now!

What about this part, where the code failed to create a symbolic link? Why does it need to create a symbolic link?
`ln: failed to create symbolic link '/lfs/h2/emc/stmp/alicia.bentley/evs_test/prod/tmp/jevs_wafs_atmos_plots.209931534.cbqs01/datainput/GFS_1P25_uvt/./evs.stats.wafs.atmos.grid2grid_uvt1p25.v20250623.stat'`

@ShannonShields-NOAA @SamiraArdani-NOAA

Commit:
To avoid 'ln: failed to create symbolic link', use '-f' (force) in 'ln'

Commit:
In EVSv2, wgrib2/2.0.8 is applied which can interpret 'ICESEV' correctly,
so MET_GRIB_TABLES (pointing to parm/metplus_config/stats/wafs/atmos/grib2_wafs.txt)
is not needed anymore.

Commit:
Corret subprocess related calls in ush/wafs/evs_wafs_atmos_plot_prune_stat_files.py

Commit:
Merge remote-tracking branch 'upstream/develop' into wdir

YaliMao-NOAA commented:
> @YaliMao-NOAA Should I point COMIN to `/lfs/h2/emc/vpppg/noscrub/yali.mao/evs/v2.0/stats/wafs` to test the plots job? Otherwise, I don't think we'll have any wind direction stats to plot! Please let me know. Thanks!

@AliciaBentley-NOAA The data quoted is still good for testing the plot job. I fixed the symbolic link issue. Also removed MET_GRIB_TABLES definition (parm/metplus_config/stats/wafs/atmos/grib2_wafs.txt) and corrected 'subprocess' related calls.

AliciaBentley-NOAA commented on parm/evs_config/wafs/config.evs.wafs.standalone:
```diff
@@ -39,22 +36,39 @@ elif [ $OBSERVATION = "GFS" ] ; then
+
+	    # 10 knots == 5.14444 m/s
+	    export VAR_OPTIONS_FCST="wind_thresh = [ >=5.14444 ]; wind_logic = INTERSECTION"
+#	    export VAR_OPTIONS_OBS="wind_thresh = [ NA ]; wind_logic = INTERSECTION"
```
@YaliMao-NOAA Can this commented out line please be removed?

YaliMao-NOAA replied:
@AliciaBentley-NOAA I removed the commented out line

AliciaBentley-NOAA commented on ush/wafs/evs_wafs_atmos_stats_preparedata.sh:
```diff
@@ -203,6 +203,9 @@ fi
 export valid_beg=$cc
 export valid_end=$cc
 if [ $runMETplus = yes ] ; then
+    # FCST_GRID_STAT_INPUT_DIR=$GRID_STAT_INPUT_BASE
+    # GRID_STAT_OUTPUT_DIR=$STATSDIR
```
@YaliMao-NOAA Can you please either 1) remove these two commented out lines or 2) add another comment above these lines that describes what they tell us? Thanks!

YaliMao-NOAA replied:
@AliciaBentley-NOAA I added another comment to explain the two coments.

AliciaBentley-NOAA commented:
@YaliMao-NOAA Thank you for making your latest updates! We really appreciate it! Since we haven't touched this PR since in over a month, I'd like to retest both WAFS stats and plots for us. Before I do that, can you please make one more addition to this PR?

In the emails that get sent to us when there is a missing WAFS file (like the emails we got today around 5am), I've noticed that the path to the missing WAFS file is not included. Did the path to the missing WAFS file get added to emails in this PR? If it did, there is nothing left to do! If it did not get added in this PR, can you please add the path to the missing forecasts in the WAFS stats missing file emails?
 
Current emails look like this:
```
 WARNING: A subset of blend forecasts ( F06) is missing for GCIP valid date 2025082012. METplus continues.
Job ID: jevs_wafs_stats.214968560.cbqs01
```

It would be helpful if they looks like this:
```
WARNING: A subset of blend forecasts (F06) is missing for GCIP valid date 2025082012. METplus continues.
Path to file: /lfs/h1/ops/prod/com/wafs/v7.0/wafs.YYYYMMDD/HH/ (Please add the correct path here! I'm just guessing!)
Job ID: jevs_wafs_stats.214968560.cbqs01
```

Thank you!

CC @AndrewBenjamin-NOAA @SamiraArdani-NOAA @ShannonShields-NOAA

AliciaBentley-NOAA commented:
@YaliMao-NOAA And if you could please re-sync your fork, that would be great! We've upgraded the develop branch to use the prod versions of MET-12.0.1/METplus-6.0.0, and I'd like to test your PR with these updates. Thanks!

Commit:
Merge remote-tracking branch 'upstream/develop' into wdir

Commit:
Add path to missing forecast files to email list

YaliMao-NOAA commented:
@AliciaBentley-NOAA Thank you for the suggestion. I added the path to the missing forecast files to email.
I also re-synchronized my fork.

Commit:
Update comments on some comments

AliciaBentley-NOAA commented:
@YaliMao-NOAA I have pulled in your latest updates and started retesting the WAFS stats job! You can find the information about my test below. The stats job is still running, so please check back again when the job completes to assess the log file and the stats output. Thanks!

### jevs_wafs_atmos_stats.sh
**Log file:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/pr774test/EVS/dev/drivers/scripts/stats/wafs/jevs_wafs_atmos_stats.o215139017
**DATA:** /lfs/h2/emc/stmp/alicia.bentley/evs_test/prod/tmp/jevs_wafs_atmos_stats.215139017.cbqs01/
**Stats:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/evs/v2.0/stats/wafs/

CC @ShannonShields-NOAA @SamiraArdani-NOAA @AndrewBenjamin-NOAA

YaliMao-NOAA commented:
@AliciaBentley-NOAA Your test run looks good to me.
The log file looks good to me.
The DATA folder was deleted but I ran the case on my own, stat files on both sides are identical, and the test run sent me email of the missing forecast files with path.

AliciaBentley-NOAA commented:
@YaliMao-NOAA Excellent! Can you please forward me the missing file email that you go from my test run? I don't think I got anything!

YaliMao-NOAA commented:
@AliciaBentley-NOAA I just forwarded one of the emails to you. Sorry I removed you from the email list on purpose for testing the missing forecast files.

AliciaBentley-NOAA commented:
@YaliMao-NOAA I have taken a look at the log file and I think that it looks good. WARNINGS are consistent with the WAFS files that were missing. I can also see the full path to the missing files now in the log file, which is good! Walltimes are consistent with EE2 standards (especially if things every run a little longer). 

I can move on to testing WAFS plots now! Where should I point COMIN to for the plots test? I can point to the emc.vpppg parallel, but we won't have any wind direction stats to make a plot with. Any suggestions?

CC @ShannonShields-NOAA

```
WARNING: missing forecast /lfs/h1/ops/prod/com/wafs/v7.0/wafs.20250820/06/grib2/0p25/blending/WAFS_0p25_blended_2025082006f018.grib2
0 + echo 'WARNING: missing forecast /lfs/h1/ops/prod/com/wafs/v7.0/wafs.20250820/06/grib2/0p25/blending/WAFS_0p25_blended_2025082006f018.grib2'
0 + echo 'WARNING: A subset of blend forecasts ( F18) is missing for GCIP valid date 2025082100. METplus continues.'
WARNING: missing forecast /lfs/h1/ops/prod/com/wafs/v7.0/wafs.20250820/06/grib2/0p25/blending/WAFS_0p25_blended_2025082006f021.grib2
0 + echo 'WARNING: missing forecast /lfs/h1/ops/prod/com/wafs/v7.0/wafs.20250820/06/grib2/0p25/blending/WAFS_0p25_blended_2025082006f021.grib2'
0 + echo 'WARNING: A subset of blend forecasts ( F21) is missing for GCIP valid date 2025082103. METplus continues.'
WARNING: missing forecast /lfs/h1/ops/prod/com/wafs/v7.0/wafs.20250820/06/grib2/0p25/blending/WAFS_0p25_blended_2025082006f024.grib2
0 + echo 'WARNING: missing forecast /lfs/h1/ops/prod/com/wafs/v7.0/wafs.20250820/06/grib2/0p25/blending/WAFS_0p25_blended_2025082006f024.grib2'
0 + echo 'WARNING: A subset of blend forecasts ( F24) is missing for GCIP valid date 2025082106. METplus continues.'
WARNING: missing forecast /lfs/h1/ops/prod/com/wafs/v7.0/wafs.20250820/06/grib2/0p25/blending/WAFS_0p25_blended_2025082006f027.grib2
0 + echo 'WARNING: missing forecast /lfs/h1/ops/prod/com/wafs/v7.0/wafs.20250820/06/grib2/0p25/blending/WAFS_0p25_blended_2025082006f027.grib2'
0 + echo 'WARNING: A subset of blend forecasts ( F27) is missing for GCIP valid date 2025082109. METplus continues.'
WARNING: missing forecast /lfs/h1/ops/prod/com/wafs/v7.0/wafs.20250820/06/grib2/0p25/blending/WAFS_0p25_blended_2025082006f030.grib2
0 + echo 'WARNING: missing forecast /lfs/h1/ops/prod/com/wafs/v7.0/wafs.20250820/06/grib2/0p25/blending/WAFS_0p25_blended_2025082006f030.grib2'
0 + echo 'WARNING: A subset of blend forecasts ( F30) is missing for GCIP valid date 2025082112. METplus continues.'
WARNING: missing forecast /lfs/h1/ops/prod/com/wafs/v7.0/wafs.20250820/06/grib2/0p25/blending/WAFS_0p25_blended_2025082006f033.grib2
0 + echo 'WARNING: missing forecast /lfs/h1/ops/prod/com/wafs/v7.0/wafs.20250820/06/grib2/0p25/blending/WAFS_0p25_blended_2025082006f033.grib2'
1 + echo 'WARNING: A subset of blend forecasts ( F33) is missing for GCIP valid date 2025082115. METplus continues.'
WARNING: missing forecast /lfs/h1/ops/prod/com/wafs/v7.0/wafs.20250820/06/grib2/0p25/blending/WAFS_0p25_blended_2025082006f036.grib2
0 + echo 'WARNING: missing forecast /lfs/h1/ops/prod/com/wafs/v7.0/wafs.20250820/06/grib2/0p25/blending/WAFS_0p25_blended_2025082006f036.grib2'
0 + echo 'WARNING: A subset of blend forecasts ( F36) is missing for GCIP valid date 2025082118. METplus continues.'
```

```
### Job 215139017.cbqs01 - Exit status is 0
    resources_used.walltime = 00:10:07
    ctime = Fri Aug 22 15:22:56 2025
    Error_Path = clogin08.cactus.wcoss2.ncep.noaa.gov:/lfs/h2/emc/vpppg/noscrub
    mtime = Fri Aug 22 15:33:10 2025
    Resource_List.walltime = 00:25:00
#PBS -l walltime=00:25:00
```

YaliMao-NOAA commented:
@AliciaBentley-NOAA You can still use stat data at /lfs/h2/emc/vpppg/noscrub/yali.mao/evs/v2.0/stats/wafs. You may experience the same warning messages as on July 9 for the same reason that those warnings come from no icing over some subregions (there are 10 subregions in WAFS EVS) on some vertical levels, considering it's summer time and the plotting is over a few days.

AliciaBentley-NOAA commented:
@YaliMao-NOAA I have kicked off the WAFS plots job and pointed COMIN to your stat files. You can find the information about the job at the paths below. The job is currently stuck in the queue, so I can't get a log file name for you yet, sorry! Please check back when the job completes to review the log file and output. Please also share a link to the plots that are produced so that we can view them online. Thanks!

### jevs_wafs_atmos_plots.sh
**Log file:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/pr774test/EVS/dev/drivers/scripts/plots/wafs/
**DATA:** /lfs/h2/emc/stmp/alicia.bentley/evs_test/prod/tmp/
**Plots:** /lfs/h2/emc/ptmp/alicia.bentley/evs/v2.0/plots/wafs/

CC @ShannonShields-NOAA

YaliMao-NOAA commented:
@AliciaBentley-NOAA The log file is jevs_wafs_atmos_plots.o215497577. It looks good to me. And the plottings look good to me too. I uploaded the plottings to rzdm para EVS, https://www.emc.ncep.noaa.gov/users/verification/aviation/wafs/para/. Only 90 days is available.

AliciaBentley-NOAA commented:
@YaliMao-NOAA Thanks! I see the warnings messages that you mentioned:

```
08/25 19:09:06.300 (evs_wafs_atmos_plot_fbias.py:363) WARNING: Could not find (and cannot plot) fbias stats for ICESEV at any threshold. Continuing ...
```

There are other warnings as well, but if the results look OK to you (and because WAFS plots are not in ops EVS), I think we can accept them for now!

```
08/25 19:09:06.365 (evs_wafs_atmos_plot_roc_curve.py:501) WARNING: blend data were all NaNs and will not be plotted.
08/25 19:09:06.365 (evs_wafs_atmos_plot_roc_curve.py:507) WARNING: Could not find (and cannot plot) farate and/or pod stats for ICESEV at any threshold. 
08/25 19:09:06.365 (evs_wafs_atmos_plot_roc_curve.py:511) WARNING: This may occur if no forecast or observed events were counted at any threshold for any model, so that all performance statistics are undefined. Continuing ...
```

One thing I do request that you update is the plots job walltime. This ~3 minute job currently requests 1 hour. Can you please change the walltime to something shorter, like 15 minutes? Thank you!

```
    resources_used.walltime = 00:02:51
    Resource_List.walltime = 01:00:00
```

CC @ShannonShields-NOAA

Commit:
Change walltime to 15 minutes

Commit:
Merge remote-tracking branch 'upstream/develop' into wdir

YaliMao-NOAA commented:
@AliciaBentley-NOAA  I changed the wall time to 15 minutes. It usually takes more than 10 minutes when data of all 90 days is available.
I also re-sync my branch.

AliciaBentley-NOAA reviewed (approved):
I have reviewed the coding changes made in this PR and I think that they look good. Log files had only expected errors from running icing verification in the summer. Emails now include the path to missing files. Walltimes were updated to meet NCO standards. WAFS plots are not in ops EVS. I approve this PR to be merged. 

CC @YaliMao-NOAA @ShannonShields-NOAA

ShannonShields-NOAA commented on dev/drivers/scripts/plots/wafs/jevs_wafs_atmos_plots.sh:
```diff
@@ -3,7 +3,7 @@
 #PBS -q dev
 #PBS -A VERF-DEV
-#PBS -l walltime=01:00:00
+#PBS -l walltime=0:15:00
```
Does this change also need to be made in the associated ecf script? Or maybe not due to not running in ops?
@AliciaBentley-NOAA @YaliMao-NOAA

AliciaBentley-NOAA replied:
@ShannonShields-NOAA Good catch. It should be changed in the .ecf script as well. Thanks! CC @YaliMao-NOAA

ShannonShields-NOAA reviewed (approved):
@YaliMao-NOAA and @AliciaBentley-NOAA I have reviewed the script changes and testing was successful. After my comment is addressed, I approve this PR to be merged.

AliciaBentley-NOAA reviewed (approved):
There does not appear to be a WAFS plots .ecf script because plots are not run in operations. We can submit this PR in its current form. Yali's PR is still synced. I will merge in a minute. Thanks!
